executor/show.go

```diff
@@ -2024,8 +2025,20 @@ func (e *ShowExec) fetchShowSessionStates(ctx context.Context) error {
 	if err = stateJSON.UnmarshalJSON(stateBytes); err != nil {
 		return err
 	}
-	// This will be implemented in future PRs.
-	tokenBytes, err := gjson.Marshal("")
+	// session token
+	var token *sessionstates.SessionToken
+	// In testing, user may be nil.
+	if user := e.ctx.GetSessionVars().User; user != nil {
+		// The token may be leaked without TLS, so we enforce a TLS connection.
+		if e.ctx.GetSessionVars().TLSConnectionState == nil {
+			//return ErrCannotMigrateSession.GenWithStackByArgs("the token must be queried in a TLS connection")
+			return errors.New("the token must be queried in a TLS connection")
```

Comment:
I think we could use `ErrSecureTransportRequired` for this error?

Reply:
The error message of `ErrSecureTransportRequired ` is `Connections using insecure transport are prohibited while --require_secure_transport=ON.`, but the `--require_secure_transport` is `OFF` when it comes here.

Reply:
Got it, thanks!

Please make the first letter upper-cased: "The token must be queried in a TLS connection"

Reply:
> Yes, but only if we require secure transport, not if we require TLS.
> 
> secure transport is either TLS or a UNIX socket.
> 
> I think allowing a UNIX socket or TLS is the right option here.

To judge whether it's a Unix socket, I made use of `SessionVars.ConnectionInfo`, which is only used for audit logs before. Thus, every connection has a `ConnectionInfo` now, which occupies a little more memory. The commit: https://github.com/pingcap/tidb/pull/36152/commits/40735d3200a35ff06a7b0516222a58bdc4a4fdf3